### ISSUE TITLE
`pwd` builtin does not respect symbolic links

### ISSUE THREAD

**Issue by jyn514** [oilshell/oil#455]
Steps to replicate:

```
(osh) ~ ▶️ mkdir physical
(osh) ~ ▶️ ln -s physical symbolic
(osh) ~ ▶️ cd symbolic/
(osh) ~/symbolic ▶️ echo $PWD
/home/joshua/symbolic
(osh) ~/symbolic ▶️ pwd
```

Expected behavior: `pwd` prints '/home/joshua/symbolic' (like bash).
Actual behavior: `pwd` prints '/home/joshua/physical'.

**Comment by andychu** [comment]
Wow this bug led me down a Unix rabbit hole ... !

- The underlying getcwd() libc call only returns the physical directory.
- What the shell does is "remember" the STRING that you passed to "cd" for the "logical" directory
- But what happens on the FIRST invocation of the shell and `pwd`?  There has been no `cd` command yet, so the shell has no string to remember.
  - It INHERITS it from the parent shell, because $PWD is exported.

But who sets $PWD in the first shell?  I don't know.  I think it's /etc/passwd.  So the login program must be setting PWD to the string in /etc/passwd so all shells can inherit it?

**Comment by jyn514** [comment]
I've been using osh as a login shell and `$PWD` is being set, so whatever program is in charge, it's not the shell.

It seems that osh already has the proper 'remember' behavior because `$PWD` shows the symbolic path, not the physical one. Somehow it's getting turned into an absolute path in the `pwd` builtin.

**Comment by andychu** [comment]
Yeah I'm not sure who sets it in the first place, but it doesn't matter.  I wrote some pretty exhaustive tests and OSH should behave just like other shells, and pwd and $PWD should be consistent.

The weird thing is that they are separate bits of state -- one that the user can modify and one that they can't!

**Comment by andychu** [comment]
Released with 0.7.pre3: https://www.oilshell.org/release/0.7.pre3/
